nuget/spec/dependabot/nuget/version_spec.rb

```diff
@@ -43,15 +43,31 @@
     end
 
     context "with an invalid version" do
-      let(:version_string) { "bad" }
+      let(:invalid_versions) do
+        %w(
+          1.2.3-0123
+          1.1.2+.123
+          1.2.3.DEV
+          -beta10
+          +beta.1
+          alpha.1
+          1.0.0-alpha_beta
+          1.0.0-alpha.......1
+          9.8.7+meta+meta
+        )
+      end
 
-      it { is_expected.to be(false) }
+      it "returns false" do
```

Comment:
This test fails as well. In the current implementation 1.2.3-0123, 1.1.2+.123 and 1.2.3.DEV are considered correct but based on semver2 and this [nuget tool](https://nugettools.azurewebsites.net/6.11.1/parse-version), they should not be.

Reply:
I'm not sure what sources that tool is using, but I've run each of these through the [official `NuGetVersion.Parse` method](https://github.com/NuGet/NuGet.Client/blob/8694ecdd055f0d0453974b0b9d0ad18926996679/src/NuGet.Core/NuGet.Versioning/NuGetVersionFactory.cs#L23) and all are invalid, so the examples you listed above are indeed bugs.